components/kubeflow/kfserving/src/kfservingdeployer.py

```diff
@@ -0,0 +1,105 @@
+        # return V1alpha1ModelSpec(custom=V1alpha1CustomSpec(container=containerSpec))
+        raise("Custom framework is not yet implemented")
+    else:
+        raise("Error: No matching framework: " + framework)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--model-name', type=str, help='Name to give to the deployed model', default="")
+    parser.add_argument('--default-model-uri', type=str, help='Path of the S3, GCS or PVC directory containing default model.')
+    parser.add_argument('--canary-model-uri', type=str, help='Optional path of the S3, GCS or PVC directory containing canary model.', default="")
+    parser.add_argument('--canary-model-traffic', type=str, help='Optional Traffic to be sent to the default model', default='0')
+    parser.add_argument('--pvc-name', type=str, help='Optional PersistentVolumeClaim to use.', default="")
+    parser.add_argument('--namespace', type=str, help='Kubernetes namespace where the KFServing service is deployed.', default='kubeflow')
+    parser.add_argument('--framework', type=str, help='Model Serving Framework', default='tensorflow')
+    parser.add_argument('--output_path', type=str, help='Path to store URI output')
+    args = parser.parse_args()
+
+    model_name = args.model_name
+    default_model_uri = args.default_model_uri
+    canary_model_uri = args.canary_model_uri
+    canary_model_traffic = args.canary_model_traffic
+    pvc_name = args.pvc_name
+    namespace = args.namespace
+    framework = args.framework.lower()
+    output_path = args.output_path
+
+    default_model_spec = ModelSpec(framework, default_model_uri)
+    metadata = client.V1ObjectMeta(name=model_name, namespace=namespace)
+
+    # Create Canary deployment if canary model uri is provided.
+    if canary_model_uri:
+        canary_model_spec = ModelSpec(framework, canary_model_uri)
+        kfsvc = V1alpha1KFService(api_version=constants.KFSERVING_GROUP + '/' + constants.KFSERVING_VERSION,
+                                  kind=constants.KFSERVING_KIND,
+                                  metadata=metadata,
+                                  spec=V1alpha1KFServiceSpec(default=default_model_spec,
+                                                             canary=canary_model_spec,
+                                                             canary_traffic_percent=int(canary_model_traffic)))
+    else:
+        kfsvc = V1alpha1KFService(api_version=constants.KFSERVING_GROUP + '/' + constants.KFSERVING_VERSION,
+                                  kind=constants.KFSERVING_KIND,
+                                  metadata=metadata,
+                                  spec=V1alpha1KFServiceSpec(default=default_model_spec))
+    KFServing = KFServingClient()
+    #TODO: Implement patch update once #277 issue is resolved
+    try:
+        KFServing.get(model_name, namespace=namespace)
+        KFServing.delete(model_name, namespace=namespace)
```

Comment:
Only try-catch the delete operation here and leave create operation to raise exception if it fails. We want to ignore delete errors and reveal create error in logs.

Reply:
Thanks - removed try-catch, and after internal discussions, we would like the delete operation logs also to be pushed out, if it goes wrong.